Use fast varint read path in LongInputStreamV1

oerling opened the pull request:
Do not decode varints that are skipped over and do not maintain a
buffer of possibly unused values. Use the OrcInputStream varint
methods to read or skip only what is needed.

highker commented on presto-orc/src/main/java/com/facebook/presto/orc/stream/LongInputStreamV1.java:
```diff
@@ -41,8 +40,7 @@ public LongInputStreamV1(OrcInputStream input, boolean signed)
         lastReadInputCheckpoint = input.getCheckpoint();
```
`lastReadInputCheckpoint` is never used anymore. Could you remove?

highker commented on presto-orc/src/main/java/com/facebook/presto/orc/stream/LongInputStreamV1.java:
```diff
@@ -82,15 +77,16 @@ public long next()
         }
         if (repeat) {
-            result = literals[0] + (used++) * delta;
+            result = repeatBase + (used) * delta;
```
remove (): `used * delta`

highker commented on presto-orc/src/main/java/com/facebook/presto/orc/stream/LongInputStreamV1.java:
```diff
@@ -23,12 +23,11 @@
-    private final long[] literals = new long[MAX_LITERAL_SIZE];
-    private int numLiterals;
+    private long repeatBase;
+    private int numInRun;
```
Could you give a more meaningful name for this?

oerling commented:
I addressed the comments. I numInRun -> numValuesInRun.  This was in Bhavani’s last verifier run and did not make new bugs.

 

From: James Sun <notifications@github.com> 
Sent: Saturday, October 26, 2019 12:55 AM
To: prestodb/presto <presto@noreply.github.com>
Cc: oerling <erling@xs4all.nl>; Author <author@noreply.github.com>
Subject: Re: [prestodb/presto] Use fast varint read path in LongInputStreamV1 (#13607)

 

@highker commented on this pull request.

minor comment only; otherwise LGTM

  _____  

In presto-orc/src/main/java/com/facebook/presto/orc/stream/LongInputStreamV1.java <https://github.com/prestodb/presto/pull/13607#discussion_r339290994> :

> @@ -41,8 +40,7 @@ public LongInputStreamV1(OrcInputStream input, boolean signed)
         lastReadInputCheckpoint = input.getCheckpoint();

lastReadInputCheckpoint is never used anymore. Could you remove?

  _____  

In presto-orc/src/main/java/com/facebook/presto/orc/stream/LongInputStreamV1.java <https://github.com/prestodb/presto/pull/13607#discussion_r339291030> :

>          }
         if (repeat) {
-            result = literals[0] + (used++) * delta;
+            result = repeatBase + (used) * delta;

remove (): used * delta

  _____  

In presto-orc/src/main/java/com/facebook/presto/orc/stream/LongInputStreamV1.java <https://github.com/prestodb/presto/pull/13607#discussion_r339291052> :

>  
     private final OrcInputStream input;
     private final boolean signed;
-    private final long[] literals = new long[MAX_LITERAL_SIZE];
-    private int numLiterals;
+    private long repeatBase;
+    private int numInRun;

Could you give a more meaningful name for this?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub <https://github.com/prestodb/presto/pull/13607?email_source=notifications&email_token=AKPPPT6XZTICVF56GCIHZ7TQQPZUBA5CNFSM4JFFZFZKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCJKFEWY#pullrequestreview-307515995> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AKPPPT4KXYGXNUXFHKUMC3TQQPZUBANCNFSM4JFFZFZA> .
